### ISSUE TITLE
Option to skip stack when attempting to play a sorcery-speed card during turn while abilities are on the stack

### ISSUE THREAD

**Issue by the-red-lily** [magefree/mage#9143]
Let's say there is a trigger or spell on the stack and I have priority. I attempt to play a new creature, land, artifact, whatever. It may signal implicitly that I'd like to pass priority until I am able to take the given action. However, to avoid passing priority erroneously, a popup or dropdown would appear. It could also be a setting in preferences.

The effect would work similarly to f10, in that you have the option of stopping when another item is added on the stack

The effect would only be enabled during phases when you'd be allowed to play the given spell (i.e. not during opponent's main phase)

**Comment by Alex-Vasile** [comment]
Is the idea that trying to cast a non-instant spell on your turn automatically activated "skip stack"?

**Comment by Alex-Vasile** [comment]
> The main idea is trying to cast a non-instant spell gives the user an option to easily "skip stack". Or, optionally it could be used to automatically skip stack if enabled in preferences

This would require a new way to show non-instant spells as castable as instant during your turn.

What is inadequate about pressing "skip stack" manually?

**Comment by the-red-lily** [comment]
The idea is to make the interface more intuitive. If I want to play 2 creatures this turn, I cast the first and pay for it, then try to cast the second one and nothing happens, and there is no feedback from the game, it causes a lot of friction and confusion. Especially for a player new to xmage.

Some of this could be resolved with some extra auto-skipping, but those advanced features are not necessarily intuitive.

I don't think showing other spells as castable is required. Just some sort of feedback to indicate why it isn't working (there is something on the stack) and an action the user can take (skip the stack, or cast an instant, or pass priority)

**Comment by Alex-Vasile** [comment]
> The idea is to make the interface more intuitive. If I want to play 2 creatures this turn, I cast the first and pay for it, then try to cast the second one and nothing happens, and there is no feedback from the game, it causes a lot of friction and confusion. Especially for a player new to xmage.
> 
> Some of this could be resolved with some extra auto-skipping, but those advanced features are not necessarily intuitive.
> 
> I don't think showing other spells as castable is required. Just some sort of feedback to indicate why it isn't working (there is something on the stack) and an action the user can take (skip the stack, or cast an instant, or pass priority)

It sounds like the actual issue you're describing is that you feel the current UI does not make it clear enough that there is something on the stack, and so you've found the people new to XMage are unsure of why they can't cast another non-instant spell?

**Comment by the-red-lily** [comment]
Yes, I suppose. The stack is off to the side rather than being center of screen. And extra contrast or highlighting could be helpful too.

I still think it wouldn't hurt to add extra ways to do a shortcut. There's no reason to have only one way to do the same thing (skip the stack). Software frequently allows multiple ways of accomplishing the same end

**Comment by sumsarz** [comment]
Why would people not aware of the stack be aware of this optional way of handling the stack?

**Comment by the-red-lily** [comment]
I had this problem, and I am aware of the existence of the stack. I think I cast a spell and it resolved, causing triggers. I didn't expect to have to manually pass priority on those triggers as in IRL magic it would be handled with nonverbal cues. And usually certain triggers have no reason to cast something in response anyway.

I still think it makes sense to at least warn a player who tries to do something they can't do (and explain why). But the most elegant solution to this problem is likely to make the stack more obvious in the UI. Right now it is on bottom right. It might make more sense on the right as a vertical stack, or in the very center (in paper magic, that's usually where the stack is represented)
